src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,652 @@
+        withdrawalRequests[requestId] = WithdrawalRequest({
+            withdrawer: msg.sender,
+            claimed: false,
+            claimTimestamp: claimTimestamp,
+            assets: SafeCast.toUint120(assets),
+            queued: queued
+        });
+
+        // burn redeemer's shares
+        _burn(msg.sender, shares);
+
+        // Remove the redeemed assets from the last available assets
+        lastAvailableAssets -= SafeCast.toInt128(SafeCast.toInt256(assets));
+
+        emit RedeemRequested(msg.sender, requestId, assets, queued, claimTimestamp);
+    }
+
+    /// @notice Claim liquidity assets from a previous withdrawal request after the claim delay has passed
+    /// @param requestId The index of the withdrawal request
+    /// @return assets The amount of liquidity assets that were transferred to the redeemer
+    function claimRedeem(uint256 requestId) external returns (uint256 assets) {
+        // Load the structs from storage into memory
+        WithdrawalRequest memory request = withdrawalRequests[requestId];
+
+        require(request.claimTimestamp <= block.timestamp, "Claim delay not met");
+        // Is there enough liquidity to claim this request?
+        require(
+            request.queued <= withdrawsClaimed + IERC20(liquidityAsset).balanceOf(address(this)),
+            "Queue pending liquidity"
+        );
+        require(request.withdrawer == msg.sender, "Not requester");
+        require(request.claimed == false, "Already claimed");
+
+        // Store the request as claimed
+        withdrawalRequests[requestId].claimed = true;
+        // Store the updated claimed amount
+        withdrawsClaimed += request.assets;
+
+        assets = request.assets;
+
+        emit RedeemClaimed(msg.sender, requestId, assets);
+
+        // transfer the liquidity asset to the withdrawer
+        IERC20(liquidityAsset).transfer(msg.sender, assets);
+    }
+
+    /// @notice Check if a withdrawal request can be claimed
+    /// @param requestId The index of the withdrawal request
+    function isClaimable(uint256 requestId) public view returns (bool) {
+        return
+            withdrawalRequests[requestId].queued <= withdrawsClaimed + IERC20(liquidityAsset).balanceOf(address(this));
+    }
+
+    /// @dev Calculate how much of the liquidity asset (WETH) in the ARM is not reserved for the withdrawal queue.
+    // That is, the amount of liquidity assets (WETH) that is available to be swapped.
+    // If there are no outstanding withdrawals, then return the maximum uint256 value.
+    // The ARM can swap out liquidity assets (WETH) that has been accrued from the performance fee for the fee collector.
+    // There is no liquidity guarantee for the fee collector. If there is not enough liquidity assets (WETH) in
+    // the ARM to collect the accrued fees, then the fee collector will have to wait until there is enough liquidity assets.
+    function _liquidityAvailable() internal view returns (uint256) {
```

Comment:
⚪️ Every place that uses (or should use) `_liquidityAvailable()` immediately takes the retuned value, and feeds it into a require statement to revert if the number is not enough. Beyond that, the value is never used.

I think we could make both this method and the calling methods cleaner by changing this method to `_requireLiquidityAvailable(uint256 amount)` and having this revert if it does not have enough.

Reply:
I like this change. I've pushed it